### DISCUSSION TITLE
Chart for timestamp, way of display

### DISCUSSION THREAD

**Discussion by manulsan** [graphieros/vue-data-ui#45] [Q&A] [answered]
For the time-stamp in Chart with x-axis,
Please see the time like "13:20:00" that time is over 12 oclock
![image](https://github.com/graphieros/vue-data-ui/assets/31757599/81c0ee53-c63a-47aa-b9fb-f364fb7c086f)

Time display  for 13:10:10
1. 01:10:10 AM
2. 13:10:10
Hope you understand what I mean.

Thanks

**Comment by graphieros** [comment, marked as answer]
Hi @manulsan ^^

There is no formatting made by the library for timestamps, it only displays the strings you pass to the config.

I recommend you use your own formatting function (or use a library for that).
Here is a function you can use to format your time:

```js
function convertTimestampToAmPm(timestamp) {
    const timestampRegex = /^(\d{4})\/(\d{2})\/(\d{2}) (\d{2}):(\d{2}):(\d{2})$/;

    const match = timestamp.match(timestampRegex);
    if (!match) {
        return "Invalid timestamp format. Please use 'YYYY/MM/DD HH:MM:SS'.";
    }

    const [_, year, month, day, hoursStr, minutesStr, secondsStr] = match;

    const date = new Date(`${year}-${month}-${day}T${hoursStr}:${minutesStr}:${secondsStr}`);

    let hours = date.getHours();
    const minutes = date.getMinutes();
    const seconds = date.getSeconds();

    const amPm = hours >= 12 ? 'PM' : 'AM';

    hours = hours % 12 || 12;

    const formattedMinutes = minutes < 10 ? '0' + minutes : minutes;
    const formattedSeconds = seconds < 10 ? '0' + seconds : seconds;

    const formattedTime = `${hours}:${formattedMinutes}:${formattedSeconds} ${amPm}`;

    return formattedTime;
}

console.log(convertTimestampToAmPm('2024/07/02 13:17:06')) // Result: "1:17:06 PM"
```

**Reply by manulsan** [reply]
Oh,,,
Let me check my side,

**Reply by manulsan** [reply]
Thanks,
It is my mistake while converting time to formatted string
By mistake, 
'hh:mm:ss' was used,
It is changed to 'HH:mm:ss' and it works as expected,

Sorry  about my mistake.

Thanks

**Reply by graphieros** [reply]
You are welcome ^^
